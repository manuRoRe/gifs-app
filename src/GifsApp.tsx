import React, { useState } from "react";
import { mockGifs } from "./mock-data/gifs.mock";
import { CustomHeader } from "./shared/components/CustomHeader";
import { SearchBar } from "./shared/components/SearchBar";
import { PreviousSearches } from "./gifs/components/PreviousSearches";
import { GifList } from "./gifs/components/GifList";

export const GifsApp = () => {
  const [previousTerms, setPreviousTerms] = useState(["dragon ball"]);

  const handleTermClick = (search: string) => {
    console.log(search);
  };

  const handleSearch = (query: string) => {
    query = query.toLowerCase();
    query = query.trim();
    if (query.length !== 0) {
      if (previousTerms.includes(query)) return;

      setPreviousTerms([query, ...previousTerms.slice(0, 7)]);
    }
  };

  return (
    <>
      {/* Header */}
      <CustomHeader
        title="Gifs App"
        subtitle="Encuentra los mejores gifs"
      ></CustomHeader>

      {/* Buscador */}
      <SearchBar placeHolder="Buscar Gifs" onQuery={handleSearch}></SearchBar>

      {/* Busquedas previas */}
      <PreviousSearches
        searches={previousTerms}
        onLabelClick={handleTermClick}
      ></PreviousSearches>

      {/* GIFS */}
      <GifList gifs={mockGifs}></GifList>
    </>
  );
};
