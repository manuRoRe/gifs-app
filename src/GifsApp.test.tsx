import { describe, expect, test } from "vitest";
import { render } from "@testing-library/react";
import { GifsApp } from "./GifsApp";

describe("GifsApp", () => {
  test("should render GifsApp", () => {
    const { container } = render(<GifsApp />);
    expect(container).toMatchSnapshot();
  });
});
