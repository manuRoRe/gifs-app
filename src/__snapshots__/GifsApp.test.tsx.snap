// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GifsApp > should render GifsApp 1`] = `
<div>
  <div
    class="content-center"
  >
    <h1>
      Gifs App
    </h1>
    <p>
      Encuentra los mejores gifs
    </p>
  </div>
  <div
    class="search-container"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Buscar
    </button>
  </div>
  <div
    class="previous-searches"
  >
    <h2>
      Busquedas previas
    </h2>
    <ul
      class="previous-searches-list"
    />
  </div>
  <div
    class="gifs-container"
  />
</div>
`;
